go/lib.h

```diff
@@ -23,19 +23,23 @@ typedef enum RequestErrorType {
  */
 typedef struct ConnectionResponse {
   const void *conn_ptr;
-  const char *error_message;
-  RequestErrorType error_type;
+  const char *connection_error_message;
 } ConnectionResponse;
 
 /**
  * Success callback that is called when a Redis command succeeds.
+ *
+ * `channel_address` is the address of the Go channel used by the callback to send the error message back to the caller of the command.
```

Comment:
replace address with index, which is the term we've used elsewhere.
on a more minor point, it's usually not to document in one place implementation details of another piece of code - this is the index of the operation which is completed in this callback call. the caller of the callback isn't aware of Go or channels.

Reply:
same throughout

Reply:
We don't use an index for the Go channel in the Go code. We aren't keeping an array of channels. It's actually the address of the channel. It might be accurate to re-type this as a void pointer instead and rename the variable to `channel_ptr`.

Reply:
Okay, Alex convinced me that it's probably better to just name it index with no mentions of channels at all, so I went with that instead.